docs/src/extend/shareable-configs.md

```diff
@@ -12,33 +12,44 @@ To share your ESLint configuration, create a **shareable config**. You can publi
 
 This page explains how to create and publish a shareable config.
 
+::: tip
+This page explains how to create a shareable config using the flat config format. For the deprecated eslintrc format, [see the deprecated documentation](shareable-configs-deprecated).
+:::
+
 ## Creating a Shareable Config
 
 Shareable configs are simply npm packages that export a configuration object. To start, [create a Node.js module](https://docs.npmjs.com/getting-started/creating-node-modules) like you normally would.
 
-The module name must take one of the following forms:
+While you can name the package in any way that you'd like, we recommend using one of the following conventions to make your package easier to identify:
 
 * Begin with `eslint-config-`, such as `eslint-config-myconfig`.
-* Be an npm [scoped module](https://docs.npmjs.com/misc/scope). To create a scoped module, name or prefix the module with `@scope/eslint-config`, such as `@scope/eslint-config` or `@scope/eslint-config-myconfig`.
+* For an npm [scoped module](https://docs.npmjs.com/misc/scope), name or prefix the module with `@scope/eslint-config`, such as `@scope/eslint-config` or `@scope/eslint-config-myconfig`.
 
 In your module, export the shareable config from the module's [`main`](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#main) entry point file. The default main entry point is `index.js`. For example:
 
 ```js
 // index.js
-module.exports = {
-
-    globals: {
-        MyGlobal: true
-    },
+export default [
+    {
+        languageOptions: {
+            globals: {
+                MyGlobal: true
+            }
+        }
+
+        rules: {
+            semi: [2, "always"]
+        }
 
-    rules: {
-        semi: [2, "always"]
     }
-
-};
+];
 ```
 
-Since the `index.js` file is just JavaScript, you can read these settings from a file or generate them dynamically.
+Because the `index.js` file is just JavaScript, you can read these settings from a file or generate them dynamically.
+
+::: tip
+Most of the time, you'll want to export an array of config objects from your shareable config. However, you can also export a single config object. Make sure your documentation clearly shows an example of how to use your shareable config inside of an `eslint.config.js` file to avoid user confusion.
```

Comment:
I think we should recommend exporting a single config object if possible, as it's easier to use. Exporting an array is needed when config objects have `files`, but that also shouldn't be recommended for shareable configs that are intended for a general audience because it's difficult to apply it to only a part of a project.

Reply:
Here's a related discussion:
* https://github.com/eslint/eslint/discussions/17834

In my view, the biggest argument in favor of exporting configs as arrays is that an array allows a config to switch between exporting one or more config objects as needed (e.g. adding a files/override object) without necessitating a breaking change in everyone consuming the config...

Reply:
> In my view, the biggest argument in favor of exporting configs as arrays is that an array allows a config to switch between exporting one or more config objects as needed (e.g. adding a files/override object) without necessitating a breaking change in everyone consuming the config.

I think that adding a new object to the exported array is by itself a breaking change in the shareable config package, as it changes the configuration for users that just spread the exported array into their config array, which will probably be a typical use.

Reply:
This may be less typical usage, seen in [eslint-plugin-square](https://github.com/square/eslint-plugin-square/blob/be50f478d35b153bcda43fbc81b14cd9b4c5703a/lib/config/ember.js) which is a config of configs, but I have added overrides in shared configs as bug fixes, such as disabling a rule for files it shouldn't apply to, or tweaking the configuration of a rule for particular files.

Reply:
> I think we should recommend exporting a single config object if possible, as it's easier to use. Exporting an array is needed when config objects have `files`, but that also shouldn't be recommended for shareable configs that are intended for a general audience because it's difficult to apply it to only a part of a project.

I think there are good arguments for both, but I don't think it's practical to crown a single object as the preferred way. You may want to export an array if you're extending another config (think `eslint:recommended`). I actually think that it will be more common to export an array than it will be to export an object, as so many shareable configs use `extends`.